sqlitecluster/SQLiteNode.cpp

```diff
@@ -2234,6 +2242,12 @@ void SQLiteNode::_updateSyncPeer()
         }
     }
 
+    // If we reached this point, it means that there are no other available peers to sync from, but leader
+    // was a valid choice. In this case, let's use it as the newSyncPeer.
+    if (!newSyncPeer && isLeaderValidPeer) {
```

Comment:
I don't think we need `isLeaderValidPeer`, can we just use this here?
```
if (!newSyncPeer && _leadPeer) {
    newSyncPeer = _leadPeer;
} 
```

Reply:
Done